cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/WorkflowHttpHandler.java

```diff
@@ -333,14 +333,13 @@ private WorkflowSpecification getWorkflowSpecForValidRun(String namespaceId, Str
     }
 
     WorkflowSpecification workflowSpec = appSpec.getWorkflows().get(workflowId);
-    ProgramId programId = new ProgramId(namespaceId, applicationId, ProgramType.WORKFLOW, workflowId);
+    ProgramId programId = new ProgramId(appId, ProgramType.WORKFLOW, workflowId);
```

Comment:
Use `ProgramId programId = appId.workflow(workflowId);` instead.

Reply:
working on this